frontend/pages/queries/QueryPage/components/QueryResults/QueryResults.tsx

```diff
@@ -213,8 +213,7 @@ const QueryResults = ({
     // TODO - clean up these conditions
     const hasNoResultsYet =
       !isQueryFinished && (!queryResults?.length || tableHeaders === null);
-    const finishedWithNoResults =
-      isQueryFinished && (!queryResults?.length || !hostsCount.successful);
```

Comment:
why is `hostsCount.successful` not considered?

Reply:
We can't use `!hostsCount.successful` to determine displaying the `finishedWithNoResults` UI because partially successful data will show up as a count for fail instead of a count for success. Instead, we can show the `finishedWithNoResults` UI by  ensuring there's no `queryResults?.length`

Example screenshot (this UI would have said "Your live query returned no results." if I didn't change line 216 beccause the `!hostCount.successful` would resolve to true so I had to change it to count `!queryResults?.length` would resolve to false since there is 1 result in `queryResults`):
<img width="1500" alt="Screenshot 2023-09-15 at 10 10 43 AM" src="https://github.com/fleetdm/fleet/assets/71795832/12acc5d2-b20f-4ae8-ad29-a28e50215d41">

Reply:
👍